File: NetSight-master/Gestione_Database/server/classes/json_message.php
<?php
    class JsonMessage
    {
        public $number_code, $message, $status, $result;
        public function __construct(Int $number_code, string $message, bool $status, $result) {
            $this->number_code = $number_code;
            $this->message = $message;
            $this->result = $result;
            $this->status = $status;
        }
    }
    
?>

File: NetSight-client/Client/Client/Classi/JsonMessage.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Client
{
    class JsonMessage
    {
        public string message;
        public int number_code;
        public bool status;
        public object? result;
        public T? GetResultObject<T>()
        {
            if (result == null) return default(T);
            return ((JObject)result).ToObject<T>();
        }
        public List<T>? GetResultArray<T>()
        {
            if (result == null) return null;
            return ((JArray)result).ToObject<List<T>>();
        }
    }
}
